llvm/include/llvm/IR/Operator.h

```diff
@@ -351,8 +352,17 @@ class FPMathOperator : public Operator {
     case Instruction::Select:
     case Instruction::Call: {
       Type *Ty = V->getType();
-      while (ArrayType *ArrTy = dyn_cast<ArrayType>(Ty))
-        Ty = ArrTy->getElementType();
+      TypeSwitch<Type *>(Ty)
```

Comment:
Please use explicit checks here. We don't really use TypeSwitch in LLVM, and I don't want it pulled into a core header.